uiowa/uiowa
# [Event feature] Events view updates

Resolves #6365 





# Todo

- [x] Move oniowa install to update hook and test
- [x] Determine a sane date range for when filter is not exposed.
- [ ] ~~Group results by date~~ https://github.com/uiowa/uiowa/pull/6530#issuecomment-1602856395
- [x] Move JS/Styles to work with all exposed filters?
- [ ] ~~Date select is not in secondary?~~ https://github.com/uiowa/uiowa/pull/6530#issuecomment-1602947319

# How to test

`ddev composer install && ddev blt ds --site=admissions.uiowa.edu && ddev drush @admissions.local uli events`
Compare https://admissions.uiowa.ddev.site/events and https://admissions.uiowa.edu/events

- There should be no changes. (Note: there may be changes from #6502 present, if that has not yet deployed)
- Check https://admissions.uiowa.ddev.site/areas-of-study to make sure the exposed form selects get the color enhancement on change.

`ddev composer install &&  ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local uli events`
Compare https://iowasummerwritingfestival.uiowa.ddev.site/events and https://iowasummerwritingfestival.uiowa.edu/events

- There should be no changes.

`ddev drush @default.local sql-drop -y && ddev drush @default.local cr && ddev blt di --site=default && ddev drush @default.local config-split:activate event && ddev drush @default.local uli admin/config/sitenow/sitenow-events`

- Don't enable the date range filter. Create some events. You should see results.
- Filter changes should be reflected immediately.

`ddev composer install && ddev blt ds --site=oniowa.uiowa.edu && ddev drush @oniowa.local uli events`
Check out https://oniowa.uiowa.ddev.site/events after creating a few events on different days.

- Date select should reflect the days from the results and work when a date is selected.
- Presenter filter should not be enabled.
- Check out the select text color changes when an option is selected.
- As a webmaster or other editor, you should see a linked message for the configuration page similar to articles and people.
